File: Proyectos/Android/ExamenFragments_DavidCarroSalinas/app/src/main/java/com/example/examenfragments_davidcarrosalinas/Modelo/Adaptador.java
package com.example.examenfragments_davidcarrosalinas.Modelo;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.examenfragments_davidcarrosalinas.R;

import java.util.ArrayList;

public class Adaptador extends ArrayAdapter {

    private ArrayList<Libro> lista;
    private Context context;
    public Adaptador(@NonNull Context context, ArrayList<Libro> libros) {
        super(context, R.layout.item, libros);
        this.context = context;
        this.lista = libros;
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        View v = LayoutInflater.from(context).inflate(R.layout.item, null);

        TextView titulo = v.findViewById(R.id.titulo);
        TextView autor = v.findViewById(R.id.autor);

        titulo.setText(lista.get(position).getTitulo());
        autor.setText(lista.get(position).getAutor());

        return v;
    }
}

File: Desarrollo multimedia y dispositivos móviles/UT2/Teoría/recursosExamenAndroidDiciembre/kotlin/Adaptador.kt
package com.example.examenfragmentskotlin

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

class Adaptador(private val context: Context, libros: ArrayList<Libro>) :
    ArrayAdapter<Libro>(context, R.layout.item, libros) {
    private val lista: ArrayList<Libro>

    init {
        lista = libros
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val v: View = LayoutInflater.from(context).inflate(R.layout.item, null)
        val titulo = v.findViewById<TextView>(R.id.titulo)
        val autor = v.findViewById<TextView>(R.id.autor)
        titulo.setText(lista[position].getTitulo())
        autor.setText(lista[position].getAutor())
        return v
    }
}
